### PR TITLE
Select default account after creation

### PR DESCRIPTION
**Pull request by cypt4**
Fixes https://github.com/brave/brave-browser/issues/22114
Default account is selected after account is created\imported for now. UI part that was selecting default acc is removed.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1) Create\import\hardware import several filecoin mainnet\testnet, solana, eth accs
2) Check that after creation of an account, proper network is choosed and last account is selected.
3) Check that last created acc is correctly selected when switch between networks.
4) Check that network is correctly selected when switch between accounts.

### CODE REVIEW

**Review by Douglashdaniel — APPROVED** [review]
++ Front-end!

File: browser/brave_wallet/keyring_service_unittest.cc [hunk 1]
```diff
@@ -2366,6 +2369,201 @@ TEST_F(KeyringServiceUnitTest, NotifyUserInteraction) {
   ASSERT_TRUE(service.IsLocked());
 }
 
+TEST_F(KeyringServiceUnitTest, SelectAddedAccount) {
+  KeyringService service(json_rpc_service(), GetPrefs());
+  ASSERT_TRUE(CreateWallet(&service, "brave"));
+  base::RunLoop().RunUntilIdle();
```
Comment (darkdh) [inline browser/brave_wallet/keyring_service_unittest.cc:2375]: What is this waiting for? It's already blocking in `CreateWallet`

File: browser/brave_wallet/ethereum_provider_impl_unittest.cc [hunk 2]
```diff
@@ -1783,6 +1783,9 @@ TEST_F(EthereumProviderImplUnitTest, SignMessageRequestQueue) {
   AddEthereumPermission(GetOrigin(), hardware);
   const std::vector<std::string> addresses = GetAddresses();
 
+  // Select account that is not participating in signing process.
+  SetSelectedAccount(addresses[1], mojom::CoinType::ETH);
```
[the remark this answers is not in the document]
Comment (darkdh) [reply browser/brave_wallet/ethereum_provider_impl_unittest.cc:1787]: Both `BraveWalletService::GetPendingSignMessageRequests` and `BraveWalletService::NotifySignMessageRequestProcessed` do not check selected account. How does it affect pending sign tasks count?

[the remark this answers is not in the document]
Comment (cypt4) [reply browser/brave_wallet/ethereum_provider_impl_unittest.cc]: I've updated the comment

**Review by darkdh — APPROVED** [review]

**Pull request merged**